tests/e2e/upgrade/node.go

```diff
@@ -0,0 +1,46 @@
+package upgrade
+
+import "github.com/ory/dockertest/v3"
+
+var baseCmd = []string{"evmosd", "start"}
+
+// Node defines evmos node params for running container
+// of specific version with custom docker run arguments
+type Node struct {
+	repository string
+	version    string
+
+	runOptions     *dockertest.RunOptions
+	withRunOptions bool
+}
+
+func NewNode(repository, version string) *Node {
```

Comment:
go doc

Reply:
Thought constructors self-descriptive enough

Reply:
it's a good practice to add comments for all functions